libarchive/plan.sh

```diff
@@ -26,7 +26,8 @@ pkg_lib_dirs=(lib)
 pkg_pconfig_dirs=(lib/pkgconfig)
 
 do_build() {
-  ./configure \
+  CFLAGS="$CFLAGS -fPIC" \
```

Comment:
Can you give a bit more background on why this is required?

The GCC docs say "This option makes a difference on the m68k, PowerPC and SPARC", but we don't run on any of those anyway.

Reply:
Sure! The issue is when you end up linking against this. If it is a PIE object, my understanding is the symbols are created with `R_X86_64_32S` relocation, which IIRC is a 32-bit compatible truncated offset for the symbols (rather than a true 64 bit). What happens is when you try to link from a true 64bit application (like for example our rust compiled libarchive crate) you'll end up with linking errors because you can't link a `R_X86_64` library to a 32-bit/signed symbol table. Check out section 4.4 of the abi docs for a breakdown: https://software.intel.com/sites/default/files/article/402129/mpx-linux64-abi.pdf

Long story short, most things should be using PIC nowadays.